activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -69,7 +69,11 @@ def select_rows(sql, name = nil, binds = [])
       end
       undef_method :select_rows
 
-      # Executes the SQL statement in the context of this connection.
+      # Executes the SQL statement in the context of this connection and returns
+      # the raw result from the connection adapter.
+      # Note: depending on your database connector, the result returned by this
+      # method may be manually memory managed. Consider using the exec_query
```

Comment:
Given we are recommending using another method because of its dangerous why keep this method as public API?